src/tokens/Assets.tsx

```diff
@@ -0,0 +1,442 @@
+  const [listFooterHeight, setListFooterHeight] = useState(0)
+
+  const scrollPosition = useSharedValue(0)
+  const footerPosition = useSharedValue(0)
+  const handleScroll = useAnimatedScrollHandler<{ prevScrollY: number }>(
+    {
+      onScroll: (event, ctx) => {
+        const scrollY = event.contentOffset.y
+        scrollPosition.value = scrollY
+
+        function clamp(value: number, min: number, max: number) {
+          return Math.min(Math.max(value, min), max)
+        }
+
+        // Omit overscroll in the calculation
+        const clampedScrollY = clamp(
+          scrollY,
+          0,
+          event.contentSize.height - event.layoutMeasurement.height
+        )
+
+        // This does the same as React Native's Animated.diffClamp
+        const diff = clampedScrollY - ctx.prevScrollY
+        footerPosition.value = clamp(footerPosition.value + diff, 0, listFooterHeight)
+        ctx.prevScrollY = clampedScrollY
+      },
+      onBeginDrag: (event, ctx) => {
+        ctx.prevScrollY = event.contentOffset.y
+      },
+    },
+    [listFooterHeight]
+  )
+
+  const animatedListHeaderStyles = useAnimatedStyle(() => {
+    if (nonStickyHeaderHeight === 0) {
+      return {
+        shadowColor: 'transparent',
+        transform: [
+          {
+            translateY: -scrollPosition.value,
+          },
+        ],
+      }
+    }
+
+    return {
+      transform: [
+        {
+          translateY:
+            scrollPosition.value > nonStickyHeaderHeight
+              ? -nonStickyHeaderHeight
+              : -scrollPosition.value,
+        },
+      ],
+      shadowColor: interpolateColor(
+        scrollPosition.value,
+        [nonStickyHeaderHeight - 10, nonStickyHeaderHeight + 10],
+        ['transparent', 'rgba(48, 46, 37, 0.15)']
+      ),
+    }
+  }, [scrollPosition.value, nonStickyHeaderHeight])
+
+  const animatedFooterStyles = useAnimatedStyle(() => {
+    return {
+      transform: [
+        {
+          translateY: footerPosition.value,
+        },
+      ],
+    }
+  }, [footerPosition.value])
+
+  useScrollAwareHeader({
+    navigation,
+    title: t('totalAssets'),
+    subtitle:
+      !tokensAreStale && totalBalanceLocal.gte(0)
+        ? t('totalBalanceWithLocalCurrencySymbol', {
+            localCurrencySymbol,
+            totalBalance: totalBalanceLocal.toFormat(2),
+          })
+        : `${localCurrencySymbol} -`,
+    scrollPosition,
+    startFadeInPosition: nonStickyHeaderHeight - HEADER_OPACITY_ANIMATION_START_OFFSET,
+    animationDistance: HEADER_OPACITY_ANIMATION_DISTANCE,
+  })
+
+  const handleMeasureNonStickyHeaderHeight = (event: LayoutChangeEvent) => {
+    setNonStickyHeaderHeight(event.nativeEvent.layout.height)
+  }
+
+  const handleMeasureListHeaderHeight = (event: LayoutChangeEvent) => {
+    setListHeaderHeight(event.nativeEvent.layout.height)
+  }
+
+  const handleMeasureListFooterHeight = (event: LayoutChangeEvent) => {
```

Comment:
This "footer" is not always visible -- just when a user can claim rewards, right? Can you share a video of the footer being shown?